docs/blog/posts/ebpf-container-challenges.md

```diff
@@ -0,0 +1,509 @@
+>> all of the same commands can also be used with `docker`.
+
+Now back on the host we have:
+
+```bash
+sudo lsns -t pid
+        NS TYPE NPROCS    PID USER      COMMAND
+4026531836 pid     337      1 root      /usr/lib/systemd/systemd rhgb --switched-root --system --deserialize 30
+4026532948 pid       1 150342 user_abcd /bin/bash
+```
+
+We can see that the host PID for the container we just started is 150342.
+
+Now let's start another container in a new shell with the same command (except
+with a different name), and run the `lsns` command again on the host.
+
+```bash
+podman run -it --name=container_2 fedora:latest /bin/bash
+```
+
+On the host:
+
+```bash
+sudo lsns -t pid
+        NS TYPE NPROCS    PID USER      COMMAND
+4026531836 pid     339      1 root      /usr/lib/systemd/systemd rhgb --switched-root --system --deserialize 30
+4026532948 pid       1 150342 user_abcd /bin/bash
+4026533041 pid       1 150545 user_abcd /bin/bash
+```
+
+We now have 3 pid namespaces -- one for root and two for the containers. Since
+we already know that the first container had PID 150342 we can conclude that the
+second container has PID 150545. However, what would we do if we didn't already
+know the PID for one of the containers?  
+
+If the container we were interested in was running a unique command, we could
+use that that disambiguate. However, in this case, both are running the same
+`/bin/bash` command.
+
+If something unique is running inside of the container, we can use the `ps -e -o
+pidns,pid,args` command to get some info.
+
+For example, run `sleep 1111` in `container_1`, then
+
+```bash
+sudo ps -e -o pidns,pid,args | grep 'sleep 1111'
+4026532948  150778 sleep 1111
+4026531836  151002 grep --color=auto sleep 1111
+```
+
+This tells us that the `sleep 1111` command is running in PID namespace
+4026532948. And,
+
+```bash
+sudo lsns -t pid | grep 4026532948
+4026532948 pid       2 150342 user_abcd /bin/bash
+```
+
+Tells us that the container's host PID is 150342.
+
+Alternatively, we could run `lsns` inside of `container_1`.
+
+```bash
+dnf install -y util-linux
+lsns -t pid
+        NS TYPE NPROCS PID USER COMMAND
+4026532948 pid       2   1 root /bin/bash
+```
+
+This tells us a few interesting things. 
+
+1. Inside the container, the PID is 1,
+1. We can't see any of the other PID namespaces inside the container.
+1. The container PID namespace is 4026532948.
+
+With the container PID namespace, we can run the `lsns -t pid | grep 4026532948`
+command as we did above to find the container's host PID
+
+Finally, the container runtime knows the pid mapping. As mentioned at the
+beginning of this section, if the unique name of the container is known, the
+following command can be used to get the host PID.
+
+```bash
+podman inspect -f '{{.State.Pid}}' container_1
+150342
+```
+
+### How bpfman Agent Finds the PID on Kubernetes
+
+The Kubernetes API for installing uprobes in containers is a bit more user
```

Comment:
This is a long sentence and I had to reread it several times. See if there is a way to break it up some?